intake_stac/catalog.py

```diff
@@ -97,21 +98,37 @@ def _load(self):
         """
         Load the STAC Catalog.
         """
-        for collection in self._stac_obj.collections():
-            self._entries[collection.id] = LocalCatalogEntry(
-                name=collection.id,
-                description=collection.title,
-                driver=StacCollection,
+        # NOTE: quick fix, logic/syntax likely could be improved!
+        lastCat = True
+
+        for subcatalog in self._stac_obj.children():
```

Comment:
I think a for-else clause would make sense here:

```python
for subcatalog in self._stac_obj.children():
    self._entries[subcatalog.id] = LocalCatalogEntry(...)
else:
    # no children
    for item in self._stac_obj.items():
        self._entries[item.id] = LocalCatalogEntry(...)
```

Reply:
This doesn't actually seem to work as expected http://python-notes.curiousefficiency.org/en/latest/python_concepts/break_else.html

Reply:
another option is to check for items:
```
items = [{'rel': 'item'}.items() <= x.items() for x in self._stac_obj._data['links']]
if any(items):
    for item in self._stac_obj.items():
        self._entries[item.id] = LocalCatalogEntry(...)
```
@matthewhanson am I missing an easier way to do this?

Reply:
i've gone with: 
```python
        subcatalog = None
        # load first sublevel catalog(s)
        for subcatalog in self._stac_obj.children():
            self._entries[subcatalog.id] = LocalCatalogEntry(...)
        if subcatalog is None:
            # load items under last catalog
            for item in self._stac_obj.items():
```